wizzard_dialog.py

```diff
@@ -0,0 +1,197 @@
+# encoding: utf-8
+# -----------------------------------------------------------
+# Copyright (C) 2017 Viktor Sklencar
+# -----------------------------------------------------------
+# Licensed under the terms of GNU GPL 2
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+# ---------------------------------------------------------------------
+
+import os
+from PyQt4 import uic
+
+from PyQt4.QtCore import *
+from PyQt4.QtGui import QStandardItemModel, QStandardItem
+
+import trigger_dialog
+from sql_generator import SqlGenerator
+
+this_dir = os.path.dirname(__file__)
+
+WIDGET, BASE = uic.loadUiType(os.path.join(this_dir, 'wizzard_dialog.ui'))
+
+
+class WizzardDialog(BASE, WIDGET):
+    def __init__(self, conn, parent=None):
+        BASE.__init__(self, parent)
+        self.setupUi(self)
+
+        self.conn = conn
+
+        self.table_model = QStandardItemModel()
+        self.tableView.setModel(self.table_model)
+
+        for schema_oid, schema_name in trigger_dialog.get_schemas(conn):
+            self.cboSourceSchema.addItem(schema_name)
+            self.cboTargetSchema.addItem(schema_name)
+
+        self.cboSourceSchema.currentIndexChanged.connect(self.populate_tables)
+        self.cboTargetSchema.currentIndexChanged.connect(self.populate_tables)
+
+        self.tableFld.textChanged.connect(self.populate_tables)
+        self.cboTablesOpt.currentIndexChanged.connect(self.populate_tables)
+
+        self.attrFld.textChanged.connect(self.populate_tables)
+        self.cboFieldsOpt.currentIndexChanged.connect(self.populate_tables)
+
+        self.layers = trigger_dialog.get_spatial_tables(conn)
+        self.populate_tables()
+
+    # TODO support for regex??
+    def filter_prefix_suffix(self, list, prefix, suffix):
+        result = []
+        ix = 0
+        for item in list:
+            if prefix and item.startswith(prefix):
+                item = item.replace(prefix, "", 1)
+            if suffix and item.endswith(suffix):
+                item = "".join(item.rsplit(suffix, 1))
+
+            result.append((item, ix))
+            ix += 1
+        return result
```

Comment:
I am quite confused how the filtering works: I would expect that only matching items would be shown in the tree view - however it seems that even non-matching tables/fields are shown...